src/android/toga_android/widgets/textinput.py

```diff
@@ -5,16 +5,37 @@
     EditText,
     Gravity,
     View__MeasureSpec,
+    TextWatcher,
 )
 from .base import Widget
 
 
+class TogaTextWatcher(TextWatcher):
+    def __init__(self, text_input_interface):
+        super().__init__()
+        self.interface = text_input_interface
+
+    def beforeTextChanged(self, _charSequence, _i, _i1, _i2):
```

Comment:
(a) I presume that these empty methods are needed to satisfy the interface, but
(b) Are the argument names required? These placeholders... aren't very illustrative.

Reply:
The argument names aren't required; I put an underscore before them to imply they're unused, and then I used the Java argument names.

The methods do have to exist, though, unless we change Rubicon-Java to allow only partially implemented interfaces, which is IMHO weird.

Reply:
The underscores are fine; is the "i1, i2" names that stand out. Is that really what they're called in the official API?

Reply:
I thought I had used the Java argument names, but now that I see your change, I don't know where my names came from. Maybe some kind of Kotlin reflection stuff that I ran into while I was prototyping in Android Studio? Anyway, thanks for fixing the names. :)